### PR TITLE
add treemap config json files

### PR DESCRIPTION
**Pull request by incandescentxxc**
I added the configuration json file to generate actual treemap classes. Before generation, I'd like to have these files to be reviewed first. Major things should be noticed include,

1. TreemapSeriesOption: breadcrumb is the combination of certain fields with `BoxLayoutOptionMixin`. I created a new file with a new type called `TreemapSeriesBreadCrumbOption`, which extends `BoxLayoutOptionMixin` and has those specific fields.
2. TreemapSeriesBreadCrumbOption: `BreadcrumbEmphasisItemStyleOption` is a newly named type that does not appear in the original codes. 
3. TreemapSeriesLevelOption: `ColorString` is actually a type of String, I used "String" directly.
4. DecalObject: `dashArrayX`, `dashArrayY` are some types of Numbers, I used corresponding Number types directly.

### CODE REVIEW

**Comment by incandescentxxc** [comment]
related to #59

**Comment by IcePear-Jzx** [comment]
Please generate Java files by running the script `config-to-java.py`. Remember to format json and java files.

**Review by IcePear-Jzx — CHANGES_REQUESTED** [review]
Please generate Java files by running the script `config-to-java.py`. Remember to format json and java files.

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-emphasis-item-style-option.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "BreadcrumbEmphasisItemStyleOption",
+  "type": "interface",
+  "extends": [""],
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-emphasis-item-style-option.json:4]: Why extending an empty string here?

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        "SeriesOption",
+        "TreemapStateOption",
+        "BoxLayoutOptionMixin",
+        "RoamOptionMixin",
+        "TreemapSeriesVisualOption"
+    ],
+    "fields": [
+        { "name": "type", "types": ["String"], "default": "treemap"},
+        { "name": "size", "types": ["Number[]", "String[]"]},
+        { "name": "sort", "types": ["Boolean"], "default": true},
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json:14]: Why does it default to `true`? Let's not set default value for any field except type. We don't generate option with fields that users don't set explicitly.

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        { "name": "clipWindow", "types": ["String"]},
+        { "name": "squareRatio", "types": ["Number"]},
+        { "name": "leafDepth", "types": ["Number"]},
+        { "name": "drillDownIcon", "types": ["String"]},
+        { "name": "zoomToNodeRatio", "types": ["Number"]},
+        { "name": "nodeClick", "types": ["String"]},
+        { "name": "breadcrumb", "types": ["TreemapSeriesBreadCrumbOption"]},
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json:21]: Let's use `TreemapSeriesBreadcrumbOption` with a lowercase `c` to keep consistent with `breadcrumb`.

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        { "name": "levels", "types": ["TreemapSeriesLevelOption"]},
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json:22]: I think it should be an array. https://github.com/apache/echarts/blob/566ba8e83dd283ba359fe9aceee0d4a12dbcd6c7/src/chart/treemap/TreemapSeries.ts#L210

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        { "name": "data", "types": ["TreemapSeriesNodeItemOption"]}
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json:23]: I think it should be an array

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json (hunk 3 above)
Comment (IcePear-Jzx) [reply scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json:21]: Looks like `TreemapSeriesBreadCrumbOption` only exists in chart/treemap, we can rename it to `BreadCrumbOption`.

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-emphasis-item-style-option.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "BreadcrumbEmphasisItemStyleOption",
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-emphasis-item-style-option.json:2]: The filename should keep consistent with the class/interface name. Please rename this file to `breadcrumb-emphasis-item-style-option.json`

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-item-style-option.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "BreadcrumbItemStyleOption",
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-item-style-option.json:2]: Please rename this file to `breadcrumb-item-style-option.json`

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-item-style-option.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  "type": "interface",
+  "extends": ["ItemStyleOption"],
+  "fields": [
+    {
+      "name": "textStyle",
+      "types": ["LabelOption"]
+    }
+  ],
+  "comments": [
+    "https://github.com/apache/echarts/blob/566ba8e83dd283ba359fe9aceee0d4a12dbcd6c7/src/chart/tree/TreeSeries.ts#L51",
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-item-style-option.json:12]: Are you sure this is the right link?

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-option.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "TreemapSeriesBreadCrumbOption",
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-breadcrumb-option.json:2]: Let's rename it to `BreadcrumbOption` with a lowercase `c`.

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-level-option.json [hunk 10]
```diff
@@ -0,0 +1,13 @@
+{
+    "name": "TreemapSeriesLevelOption",
+    "type": "interface",
+    "extends": ["TreemapSeriesVisualOption","TreemapStateOption","StatesOptionMixin"],
+    "fields": [
+        { "name": "color", "types": ["String"]},
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-level-option.json:6]: `color` can also be `String[]`

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-node-item-option.json [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+  "fields": [
+    { "name": "id", "types": ["String", "Number"] },
+    { "name": "name", "types": ["String", "Number"] },
+    { "name": "value", "types": ["Number","Number[]"] },
+    { "name": "children", "types": ["TreemapSeriesNodeItemOption[]"] },
+    { "name": "color", "types": ["String"] },
+    { "name": "decal", "types": ["DecalObject[]"] }
+  ],
+  "comments": [
+    "https://github.com/apache/echarts/blob/566ba8e83dd283ba359fe9aceee0d4a12dbcd6c7/src/chart/treemap/TreemapSeries.ts#L93",
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-node-item-option.json:18]: Wrong link?

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json (hunk 2 above)
Comment (incandescentxxc) [reply scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-series-option.json:14]: There is a comment saying the default is 'desc' which means 'true' equivaltly,  in the source codes of ECharts.
 ``` /**
     * If sort in desc order.
     * Default to be desc. asc has strange effect
     */
    sort?: boolean | 'asc' | 'desc'
```
I have no preference for this though, can set it to ["Boolean", "String"].

File: scripts/config/org/icepear/echarts/charts/treemap/treemap-emphasis.json [hunk 12]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "TreemapEmphasis",
+    "type": "class",
+    "implements": ["TreemapEmphasisOption"]
+  }
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/charts/treemap/treemap-emphasis.json:5]: Did you format it?

File: scripts/config/org/icepear/echarts/origin/chart/treemap/breadcrumb-item-style-option.json [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+  "extends": ["ItemStyleOption"],
+  "fields": [
+    {
+      "name": "textStyle",
+      "types": ["LabelOption"]
+    }
+  ],
+  "comments": [
+    "https://github.com/apache/echarts/blob/566ba8e83dd283ba359fe9aceee0d4a12dbcd6c7/src/chart/treemap/TreemapSeries.ts#L51",
+    "https://github.com/apache/echarts/blob/790687df55a5dbe286e52cf182c0983938efd367/src/util/types.ts#L1090"
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/breadcrumb-item-style-option.json:13]: Why link this URL here?

File: scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-emphasis-option.json [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+      {
+        "name": "focus",
+        "types": ["String"]
+      }
+    ],
+    "comments": [
+      "https://github.com/apache/echarts/blob/566ba8e83dd283ba359fe9aceee0d4a12dbcd6c7/src/chart/treemap/TreemapSeries.ts#L87",
+      "https://github.com/apache/echarts/blob/790687df55a5dbe286e52cf182c0983938efd367/src/util/types.ts#L1541"
+    ]
+  }
```
Comment (IcePear-Jzx) [inline scripts/config/org/icepear/echarts/origin/chart/treemap/treemap-emphasis-option.json:15]: format?

**Comment by IcePear-Jzx** [comment]
Could you create a simplified `TreeMap` like `Sunburst.java`?

**Comment by jbaron** [comment]
In case I can help by testing some SNAPSHOT release before sending it to Maven Central, just let me know.

File: scripts/config/org/icepear/echarts/origin/chart/treemap/breadcrumb-item-style-option.json (hunk 13 above)
Comment (incandescentxxc) [reply scripts/config/org/icepear/echarts/origin/chart/treemap/breadcrumb-item-style-option.json:13]: the second URL refers to the type of the field "LabelOption".
what URL should we explicitly include?

Comment (IcePear-Jzx) [reply scripts/config/org/icepear/echarts/origin/chart/treemap/breadcrumb-item-style-option.json:13]: Normally, we only include the URL describing this class/interface.
In some special cases e.g. we manually add some fields existing in other places, we can link the related URLs here.
The URL for LabelOption should be included in LabelOption.java not here.

**Comment by incandescentxxc** [comment]
> Could you create a simplified `TreeMap` like `Sunburst.java`?

added in the latest commit

File: src/test/resources/simple/treemap/basic-treemap.json [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+
```
Comment (IcePear-Jzx) [inline src/test/resources/simple/treemap/basic-treemap.json:40]: format?

File: src/test/java/org/icepear/echarts/advanced/treemap/BasicTreemapChartTest.java [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+import org.icepear.echarts.origin.chart.treemap.TreemapSeriesOption;
+import org.icepear.echarts.serializer.EChartsSerializer;
+import org.junit.Test;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+public class BasicTreemapChartTest {
+
+        @Test
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/advanced/treemap/BasicTreemapChartTest.java:20]: Looks like your formatting tool is using 8 spaces for an indent

**Review by IcePear-Jzx — APPROVED** [review]
LGTM

**Pull request merged**